besu/src/main/java/org/hyperledger/besu/controller/BesuControllerBuilder.java

```diff
@@ -604,26 +603,43 @@ private WorldStateArchive createWorldStateArchive(
   protected List<PeerValidator> createPeerValidators(final ProtocolSchedule protocolSchedule) {
     final List<PeerValidator> validators = new ArrayList<>();
 
-    final OptionalLong daoBlock = configOptionsSupplier.get().getDaoForkBlock();
-    if (daoBlock.isPresent()) {
-      // Setup dao validator
-      validators.add(
-          new DaoForkPeerValidator(protocolSchedule, metricsSystem, daoBlock.getAsLong()));
-    }
+    configOptionsSupplier
```

Comment:
This is less readable than an if block.

Reply:
I usually find the lambda version more effective to describe what to do in case of something is present, and less error prone, but I can easily read both versions, so I will revert.